pkgs/by-name/ml/mlibc/package.nix

```diff
@@ -0,0 +1,127 @@
+{
+  lib,
+  buildPackages,
+  stdenvNoLibc,
+  fetchFromGitea,
+  fetchFromGitHub,
+  fetchpatch,
+  ninja,
+  pkg-config,
+  linuxHeaders,
+}:
+
+let
+  freestnd-c-hdrs = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "osdev";
+    repo = "freestnd-c-hdrs";
+    rev = "4039f438fb1dc1064d8e98f70e1cf122f91b763b";
+    sha256 = "sha256-JMb1DS9+IDqGXdivMdy326kh+5iBniGc2RHihCfhLeA=";
+  };
+  freestnd-cxx-hdrs = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "osdev";
+    repo = "freestnd-cxx-hdrs";
+    rev = "85096df5361a4d7ef2ce46947e555ec248c2858e";
+    sha256 = "sha256-sQaIRgJgTLEci7kEJSsiGen/0jWSsyICqB5Ynk6XNu4=";
+  };
+  frigg = fetchFromGitHub {
+    owner = "managarm";
+    repo = "frigg";
+    rev = "ea0e9b5ea709b0c1b3e72ffc5de7019ca46ef427";
+    sha256 = "sha256-YmnyAdSmEJxGpvKy9pXXBWnNAY6XUcJIBBjU6VSW9Q0=";
+  };
```

Comment:
Would there be any benefit to packaging these separately?

Reply:
Packaging `freestnd-*-hdrs` separately would benefit the homebrew OSdev community that builds their OSes with Nix. For now, seemingly that is only me. I followed the thinking of "it deserves to be in nixpkgs if it would be used by more than one person" so I didn't do it but I can package them separately if that's preferred.

Packaging `frigg` separately would be helpful for building Managarm and building software targeting Managarm. I intend on doing both in nixpkgs in the future. I'll move it to a separate package

Reply:
I don't see the harm in having separate packages, and means they can potentially be auto-updated and will be less likely to end up duplicated in the event some other package needs them.